src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1193,10 +1192,12 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
         {
             Symbol? symbol = BindPropertyPatternMember(inputType, name, ref hasErrors, diagnostics);
 
-            if (inputType.IsErrorType() || hasErrors || symbol is null)
-                memberType = CreateErrorType();
-            else
-                memberType = symbol.GetTypeOrReturnType().Type;
+            memberType = symbol switch
+            {
+                FieldSymbol field => field.Type,
+                PropertySymbol property => property.Type,
+                _ => CreateErrorType()
```

Comment:
>_ [](start = 16, length = 1)

It looks like there is a possible behavior change around other kinds of symbols. If other kinds not possible, consider explicitly checking for null and throwing for everything else.

Reply:
> It looks like there is a possible behavior change around other kinds of symbols. 

There shouldn't be any change for other symbol kinds. `BindPropertyPatternMember()` sets `hasErrors` for symbols other than fields or properties and previously we used `CreateErrorType()` if `hasErrors` was set.

---
In reply to: [597137302](https://github.com/dotnet/roslyn/pull/51950#discussion_r597137302) [](ancestors = 597137302)